src/CardArea/CardArea.js

```diff
@@ -21,23 +23,52 @@ class CardArea extends Component {
   sortCards() {
     this.props.userCardsData.sort(this.sortCardsByName);
     return this.props.userCardsData.map(userCard => {
-      return <Card userCard={userCard} key={userCard.cardName} />
+      return <Card userCard={userCard} 
+                   key={userCard.cardName}
+                   cardAreaView={this.props.cardAreaView} />
     });
   }
 
   sortDecks() {
     this.props.userDecks.sort((a, b) => a.price - b.price);
     return this.props.userDecks.map(userDeck => {
-      return <Deck userDeck={userDeck} key={userDeck.deckName} />
+      return <Deck 
+                userDeck={userDeck} 
+                key={userDeck.deckName}
+                expandDeck={this.expandDeck}
+                setCardAreaView={this.props.setCardAreaView}
+              />
     });
   }
 
+  
+  expandDeck = (deckObj) => {
+    let selectedDeck = this.props.getExpandedDeckInfo(deckObj)
+    this.setState( {selectedDeck} );
+  }
+
+  displayDeck = () => {
+    if (this.state.selectedDeck === []) {
+      return null;
+    } else {
+      return this.state.selectedDeck.map(card => {
+        return <Card cardImage={card.imageSource} 
+                    price={card.price}
+                    key={card.cardName}
+                    cardAreaView={this.props.cardAreaView}/>
+      })
+    }
+  }
```

Comment:
In what case would `this.state.selectedDeck === []`?

Reply:
I removed this because it was not needed! 👍